Warn about NGINX LB being recreated with upgrade to 1.8.0+ (#435)
giantswarm/releases: announcements/2020-09-02-optional-nginx-lb-recreate-on-upgrade-to-1.8.0+.md

> ### :zap: Dear Giant Swarm customer :zap:

Reviewer: ```suggestion
## Known Issue: Potential ingress downtime when upgrading to NGINX IC v1.8.0+
```

Make title more descriptive.

> We'd like to inform you about a known issue when upgrading NGINX ingress controller (IC) optional app to v1.8.0+ on AWS and Azure clusters - NGINX IC LoadBalancer Service may get renamed/recreated with the app upgrade.

Reviewer 2: ```suggestion
We'd like to inform you about a known issue when upgrading NGINX ingress controller (IC) optional app to v1.8.0+ on *AWS and Azure* clusters - NGINX IC LoadBalancer Service may get renamed/recreated with the app upgrade.
```

Author: Fine with making it bold, still I assume this will be sent to AWS and Azure customers only.

Reviewer 2: True, but some customers have both cloud and on prem

> In older releases the Service name was hardcoded to `nginx-ingress-controller`. As of v1.8.0, to ensure its uniqueness for multiple NGINX ICs per cluster support, the LoadBalancer Service name was made to be dynamic, derived from Helm release i.e. App Custom Resource name.

Author: ```suggestion
In older releases the Service name was hardcoded to `nginx-ingress-controller`. As of v1.8.0, to ensure its uniqueness for multiple NGINX ICs per cluster support, the LoadBalancer Service name was made to be dynamic, derived from Helm release i.e. App Custom Resource (CR) name.
```

> NGINX IC LoadBalancer Service gets replaced by a new one for every NGINC IC App CR whose name is not `nginx-ingress-controller`.

Author: ```suggestion
Therefore, NGINX IC LoadBalancer Service will be replaced by a new one for every NGINC IC App CR whose name is not `nginx-ingress-controller`.
```

When NGINX IC LoadBalancer Service gets recreated, cloud service provider (CSP) load balancer behind it gets recycled as well.
It can take minute or so for ingress DNS records to be updated by external-dns and change propagated to clients.
During that time there's ingress traffic downtime, since clients still resolve old no longer present CSP load balancer.

> Please take the potential downtime into consideration when planning the upgrade.

Author: ```suggestion
** Please take the potential downtime into consideration when planning the upgrade. **
```

To make sure the downtime is shortest possible, external-dns availability is important precondition.
In recent platform releases (Azure v12.0.2, and AWS v12.1.4 and v11.5.4) we've improved external-dns monitoring and alerting.

> Therefore, before upgrading NGINX IC optional app to 1.8.0+, please make sure that your cluster has been upgraded to the latest release.

Reviewer: ```suggestion
**Therefore, before upgrading NGINX IC optional app to 1.8.0+, please make sure that your cluster has been upgraded to the latest release.**
```

Bold the actionable.

> **tl;dr: Therefore, before upgrading NGINX IC optional app to 1.8.0+, please make sure that your cluster has been upgraded to the latest release.**

Author: ```suggestion
**tl;dr: Therefore, before upgrading NGINX IC optional app to 1.8.0+, please make sure that your cluster has been upgraded to the latest platform release.**
```
